controllers/cloudstackmachinestatechecker_controller.go

```diff
@@ -0,0 +1,99 @@
+)
+
+//+kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=cloudstackmachinestatecheckers,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=cloudstackmachinestatecheckers/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=cloudstackmachinestatecheckers/finalizers,verbs=update
+//+kubebuilder:rbac:groups=cluster.x-k8s.io,resources=machines,verbs=get;list;watch;delete
+
+// CloudStackMachineStateCheckerReconciliationRunner is a ReconciliationRunner with extensions specific to CloudStackCluster reconciliation.
+type CloudStackMachineStateCheckerReconciliationRunner struct {
+	csCtrlrUtils.ReconciliationRunner
+	MachineStateCheckers  *infrav1.CloudStackMachineStateCheckerList
+	ReconciliationSubject *infrav1.CloudStackMachineStateChecker
+	CAPIMachine           *capiv1.Machine
+	CSMachine             *infrav1.CloudStackMachine
+}
+
+// CloudStackMachineStateCheckerReconciler reconciles a CloudStackMachineStateChecker object
+type CloudStackMachineStateCheckerReconciler struct {
+	csCtrlrUtils.ReconcilerBase
+}
+
+// Initialize a new CloudStackMachineStateChecker reconciliation runner with concrete types and initialized member fields.
+func NewCSMachineStateCheckerReconciliationRunner() *CloudStackMachineStateCheckerReconciliationRunner {
+	// Set concrete type and init pointers.
+	runner := &CloudStackMachineStateCheckerReconciliationRunner{ReconciliationSubject: &infrav1.CloudStackMachineStateChecker{}}
+	runner.CAPIMachine = &capiv1.Machine{}
+	runner.CSMachine = &infrav1.CloudStackMachine{}
+	// Setup the base runner. Initializes pointers and links reconciliation methods.
+	runner.ReconciliationRunner = csCtrlrUtils.NewRunner(runner, runner.ReconciliationSubject)
+	return runner
+}
+
+func (reconciler *CloudStackMachineStateCheckerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	return NewCSMachineStateCheckerReconciliationRunner().
+		UsingBaseReconciler(reconciler.ReconcilerBase).
+		ForRequest(req).
+		WithRequestCtx(ctx).
+		RunBaseReconciliationStages()
+}
+
+func (r *CloudStackMachineStateCheckerReconciliationRunner) Reconcile() (ctrl.Result, error) {
+	if res, err := r.GetParent(r.ReconciliationSubject, r.CSMachine)(); r.ShouldReturn(res, err) {
+		return res, err
+	}
+	if res, err := r.GetParent(r.CSMachine, r.CAPIMachine)(); r.ShouldReturn(res, err) {
+		return res, err
+	}
+
+	if err := r.CS.ResolveVMInstanceDetails(r.CSMachine); err != nil {
+		return r.ReturnWrappedError(err, "failed to resolve VM instance details")
+	}
+	if r.CSMachine.Status.InstanceState == "Running" {
+		r.ReconciliationSubject.Status.Ready = true
+	} else {
+		if err := r.Client.Delete(r.RequestCtx, r.CAPIMachine); err != nil {
+			return r.ReturnWrappedError(err, "failed to delete CAPI machine")
+		}
+	}
+
+	return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
```

Comment:
Should we make the requeue period customizable?

Reply:
Let's leave it as is. When we have a request to make the period customizable let's revisit this.